internal/backend/secrets/aws.go

```diff
@@ -0,0 +1,108 @@
+package secrets
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"go.uber.org/zap"
+)
+
+type awsSecrets struct {
+	client  *secretsmanager.Client
+	logger  *zap.Logger
+	timeout time.Duration
+}
+
+// NewAWSSecrets initializes a client connection to AWS Secrets Manager.
+func NewAWSSecrets(l *zap.Logger, c *Config) (Secrets, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		l.Error("AWS config initialization", zap.Error(err))
+		return nil, err
+	}
+
+	client := secretsmanager.NewFromConfig(cfg)
+	return &awsSecrets{client: client, logger: l, timeout: c.Timeout}, nil
+}
+
+// The data size limit is 64 KiB, according to this link:
+// https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_CreateSecret.html
+func (s *awsSecrets) Set(scope, name string, data map[string]string) error {
+	ctx, cancel := limitedContext(s.timeout)
```

Comment:
not sure why not just passing the context from the caller?

Reply:
1. Contexts are an implementation-specific detail, not all implementations use them
2. The context for these calls needs to be stricter (short timeout by default, even if the caller is a long-running runtime workflow)

Reply:
done